app/azure/request_adapter.py

```diff
@@ -92,8 +92,31 @@ def _messages_to_responses_input_and_instructions(
 
     def _transform_tools_for_responses(self, tools: Any) -> Any:
         out: List[Dict[str, Any]] = []
-        for tool in tools:
+        if not isinstance(tools, list):
+            current_app.logger.debug(
+                "Skipping tool transformation because tools payload is not a list: %r",
+                tools,
+            )
+            return out
+
+        for idx, tool in enumerate(tools):
```

Comment:
I don't think we need to support this branches where tools / functions are not the right type, as that means the payload is not following the standard, and we should only go as far as to support the standards instead of any possible arbitrary missuse.

When removing please change back to `for tool in tools:` as the `idx` will no longer be needed.